source/extensions/filters/network/common/redis/codec_impl.cc

```diff
@@ -138,6 +171,31 @@ RespValue::RespValue(const RespValue& other) : type_(RespType::Null) {
   }
 }
 
+RespValue::RespValue(RespValue&& other) noexcept : type_(other.type_) {
```

Comment:
Why do we need the move constructor and move assignment now? In general it's better to avoid magic methods unless we really need them.

Reply:
This is needed for the full change for 8606.

Reply:
This is needed #8606.